.editorconfig

```diff
@@ -0,0 +1,11 @@
+# EditorConfig is awesome: https://EditorConfig.org
+
+root = true
+
+[*]
+charset = utf-8
+indent_style = space
+trim_trailing_whitespace = true
+insert_final_newline = true
+end_of_line = lf
+indent_size = 2
```

Comment:
this is actually really nice to have, but should:

* get its own pull request
* be documented that it exists and [how to use it in various editors](https://editorconfig.org/#download), maybe in the readme or a `contributing.md` file
* current code is a mix between all kinds of indention styles (2 spaces, 4 spaces, tabs), it should be adjusted to adhere to these settings first.

Reply:
Sure. I will move this to a separate PR and add some documentation details.